src/server/blockchain.cpp

```diff
@@ -351,6 +355,7 @@ map<string, uint64_t> BlockChain::getHeaderChainStats() {
 }
 
 ExecutionStatus BlockChain::startChainSync() {
+    std::unique_lock<std::mutex> ul(lock);
```

Comment:
This results in deadlock since startChainSync calls addBlock that uses the same lock.